helper/threads.c

```diff
@@ -65,8 +98,14 @@ static void *run_thread(void *arg)
 		 "pthread" : "process",
 		 (int)getpid());
 
-	if (odp_atomic_load_u32(&start_args->status) == SYNC_INIT)
-		odp_atomic_store_rel_u32(&start_args->status, INIT_DONE);
+	if (start_args->status == SYNC_INIT) {
+		if (set_child_status(start_args->sockfd, INIT_DONE)) {
+			ODPH_ERR("Init status send failed\n");
```

Comment:
At this level, the log could state that "Init status *setting* failed" as the function name is `set_child_status()`, otherwise looks OK.

Reply:
Fixed in V3.